File: Android_Nativo_Java/Projeto GS300 novo/app/src/main/java/com/example/gs300_pdv_java/MainActivity.java
package com.example.gs300_pdv_java;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;

import com.example.gs300_pdv_java.fala.FalaActivity;
import com.example.gs300_pdv_java.impressora.Impressora;
import com.example.gs300_pdv_java.msitef.Tef;
import com.example.gs300_pdv_java.nfc.NfcExemplo;
import com.example.gs300_pdv_java.sat.MenuSat;
import com.example.gs300_pdv_java.sublcd.SubLcd;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private Class WangPosManagerClass;
    private Object WangPosManager;
    public static String Model = Build.MODEL;
    public static String PLATAFORMA = "Android Studio";
    public static final String VERSION = "1.0.0";

    public static String getVersion(){
        return PLATAFORMA+" - "+VERSION+" - GS300";
    }

    //REFERÊNCIA DAS ACTIVITIES
    public static final String ACTIVITY_SEGUNDO_DISPLAY = "Segundo Display";
    public static final String ACTIVITY_NFC = "NFC - NDF";
    public static final String ACTIVITY_FALA = "Texto para Fala";
    public static final String  ACTIVITY_SAT = "Sat";
    public static final String  ACTIVITY_TEF = "Tef";
    public static final String  ACTIVITY_IMPRESSORA = "Impressão";
    public static final String  ACTIVITY_TEF_FEIRA = "Tef";

    ArrayList<Projeto> projetos = new ArrayList<Projeto>();
    ListView lvProjetos;

    TextView txtProject;
    private AlertDialog alerta;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        lvProjetos = findViewById(R.id.lvProjetos);
        txtProject = findViewById(R.id.txtNameProject);
        txtProject.setText(getVersion());

        projetos.add(new Projeto(ACTIVITY_IMPRESSORA, R.drawable.print));
        projetos.add(new Projeto(ACTIVITY_SEGUNDO_DISPLAY, R.drawable.customerdisplay));
        projetos.add(new Projeto(ACTIVITY_NFC, R.drawable.nfc2));
        projetos.add(new Projeto(ACTIVITY_FALA, R.drawable.speaker));
        projetos.add(new Projeto(ACTIVITY_TEF, R.drawable.tef));
        projetos.add(new Projeto(ACTIVITY_SAT, R.drawable.icon_sat));
        //projetos.add(new Projeto(ACTIVITY_TEF_FEIRA, R.drawable.icon_sat));

        ProjetoAdapter adapter = new ProjetoAdapter(getBaseContext(), R.layout.listprojetos, projetos);
        lvProjetos.setAdapter(adapter);
        lvProjetos.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Projeto projeto = (Projeto) lvProjetos.getItemAtPosition(i);

                Intent intent = null;
                switch (projeto.getNome()){
                    //G-BOT Implementação Futura
                    case ACTIVITY_SEGUNDO_DISPLAY:
                        intent = new Intent(MainActivity.this, SubLcd.class);
                        break;
                    case ACTIVITY_NFC:
                        intent = new Intent(MainActivity.this, NfcExemplo.class);
                        break;

                    //A112
                    /*case ACTIVITY_IMPRESSAO:
                        intent = new Intent(MainActivity.this, Impressora.class);
                        break;*/
                    /*case ACTIVITY_DISPLAY:
                        intent = new Intent(MainActivity.this, DisplayActivity.class);
                        break;*/

                    //Activities Comuns
                    case ACTIVITY_TEF:
                        intent = new Intent(MainActivity.this, Tef.class);
                        break;

                    case ACTIVITY_SAT:
                        intent = new Intent(MainActivity.this, MenuSat.class);
                        break;

                    case ACTIVITY_FALA:
                        intent = new Intent(MainActivity.this, FalaActivity.class);
                        break;

                    case ACTIVITY_IMPRESSORA:
                        intent = new Intent(MainActivity.this, Impressora.class);
                        break;

                    /*case ACTIVITY_TEF_FEIRA:
                        intent = new Intent(MainActivity.this, SensorActivity.class);
                        break;*/

                }

                if(intent != null){
                    startActivity(intent);
                }
            }
        });
    }
}

File: Android_Nativo_Kotlin/app/src/main/java/com/example/gs300_2023/MainActivity.kt
package com.example.gs300_2023

import android.content.Intent
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import com.example.gs300_2023.fala.FalaActivity
import com.example.gs300_2023.impressora.ImpressoraActivity
import com.example.gs300_2023.msitef.Tef
import com.example.gs300_2023.nfc.NfcExemplo
import com.example.gs300_2023.sat.MenuSat
import com.example.gs300_2023.sublcd.SubLcd


class MainActivity : AppCompatActivity() {
    private lateinit var WangPosManagerClass: Class<*>
    private lateinit var WangPosManager: Any
    private val Model = Build.MODEL

    companion object {
        fun getVersion(): String? {
            val PLATAFORMA = "Android Studio"
            val VERSION = "1.0.0"
            return "$PLATAFORMA - $VERSION - GS300"
        }
        const val ACTIVITY_SEGUNDO_DISPLAY = "Segundo Display"
        const val ACTIVITY_NFC = "NFC - NDF"
        const val ACTIVITY_FALA = "Texto para Fala"
        const val ACTIVITY_SAT = "Sat"
        const val ACTIVITY_TEF = "Tef"
        const val ACTIVITY_IMPRESSORA = "Impressão"
        const val ACTIVITY_TEF_FEIRA = "Tef"
    }

    private var projetos = ArrayList<Projeto>()
    private lateinit var lvProjetos: ListView

    private lateinit var txtProject: TextView
    private lateinit var alerta: AlertDialog
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        lvProjetos = findViewById(R.id.lvProjetos)
        txtProject = findViewById(R.id.txtNameProject)
        txtProject.text = getVersion()
        projetos.add(Projeto(ACTIVITY_IMPRESSORA, R.drawable.print))
        projetos.add(Projeto(ACTIVITY_SEGUNDO_DISPLAY, R.drawable.customerdisplay))
        projetos.add(Projeto(ACTIVITY_NFC, R.drawable.nfc2))
        projetos.add(Projeto(ACTIVITY_FALA, R.drawable.speaker))
        projetos.add(Projeto(ACTIVITY_TEF, R.drawable.tef))
        projetos.add(Projeto(ACTIVITY_SAT, R.drawable.icon_sat))
        //projetos.add(Projeto(ACTIVITY_TEF_FEIRA, R.drawable.icon_sat))
        val adapter = ProjetoAdapter(baseContext, R.layout.listprojetos, projetos)
        lvProjetos.adapter = adapter
        lvProjetos.setOnItemClickListener { adapterView, view, i, l ->
            val projeto = lvProjetos.getItemAtPosition(i) as Projeto

            var intent: Intent? = null
            when (projeto.nome) {
                //G-BOT Implementação Futura
                ACTIVITY_SEGUNDO_DISPLAY -> {
                    intent = Intent(this@MainActivity, SubLcd::class.java)
                }
                ACTIVITY_NFC -> {
                    intent = Intent(this@MainActivity, NfcExemplo::class.java)
                }

                //A112
                /*ACTIVITY_IMPRESSAO -> {
                    intent = Intent(this@MainActivity, Impressora::class.java)
                }*/
                /*ACTIVITY_DISPLAY -> {
                    intent = Intent(this@MainActivity, DisplayActivity::class.java)
                }*/

                //Activities Comuns
                ACTIVITY_TEF -> {
                     intent = Intent(this@MainActivity, Tef::class.java)
                }

                ACTIVITY_SAT -> {
                     intent = Intent(this@MainActivity, MenuSat::class.java)
                }

                ACTIVITY_FALA -> {
                    intent = Intent(this@MainActivity, FalaActivity::class.java)
                }

                ACTIVITY_IMPRESSORA -> {
                    intent = Intent(this@MainActivity, ImpressoraActivity::class.java)
                }

//                ACTIVITY_TEF_FEIRA -> {
//                    intent = Intent(this@MainActivity, SensorActivity::class.java)
//                }
            }

            if (intent != null) {
                startActivity(intent)
            }
        }

    }

}